covid_xprize/standard_predictor/xprize_predictor.py

```diff
@@ -0,0 +1,607 @@
+# Copyright 2020 (c) Cognizant Digital Business, Evolutionary AI. All rights reserved. Issued under the Apache 2.0 License.
+
+import os
+
+# Suppress noisy Tensorflow debug logging
+os.environ['TF_CPP_MIN_LOG_LEVEL'] = '3'
+
+# noinspection PyPep8Naming
+import keras.backend as K
+import numpy as np
+import pandas as pd
+from keras.callbacks import EarlyStopping
+from keras.constraints import Constraint
+from keras.layers import Dense
+from keras.layers import Input
+from keras.layers import LSTM
+from keras.layers import Lambda
+from keras.models import Model
+
+
+ROOT_DIR = os.path.dirname(os.path.abspath(__file__))
+DATA_PATH = os.path.join(ROOT_DIR, 'data')
+DATA_FILE_PATH = os.path.join(DATA_PATH, 'OxCGRT_latest.csv')
+ADDITIONAL_CONTEXT_FILE = os.path.join(DATA_PATH, "Additional_Context_Data_Global.csv")
+ADDITIONAL_US_STATES_CONTEXT = os.path.join(DATA_PATH, "US_states_populations.csv")
+ADDITIONAL_UK_CONTEXT = os.path.join(DATA_PATH, "uk_populations.csv")
+ADDITIONAL_BRAZIL_CONTEXT = os.path.join(DATA_PATH, "brazil_populations.csv")
+
+NPI_COLUMNS = ['C1_School closing',
+               'C2_Workplace closing',
+               'C3_Cancel public events',
+               'C4_Restrictions on gatherings',
+               'C5_Close public transport',
+               'C6_Stay at home requirements',
+               'C7_Restrictions on internal movement',
+               'C8_International travel controls',
+               'H1_Public information campaigns',
+               'H2_Testing policy',
+               'H3_Contact tracing',
+               'H6_Facial Coverings']
+
+CONTEXT_COLUMNS = ['CountryName',
+                   'RegionName',
+                   'GeoID',
+                   'Date',
+                   'ConfirmedCases',
+                   'ConfirmedDeaths',
+                   'Population']
+NB_LOOKBACK_DAYS = 21
+NB_TEST_DAYS = 14
+WINDOW_SIZE = 7
+US_PREFIX = "United States / "
+NUM_TRIALS = 1
+LSTM_SIZE = 32
+MAX_NB_COUNTRIES = 20
+
+
+class Positive(Constraint):
+
+    def __call__(self, w):
+        return K.abs(w)
+
+
+# Functions to be used for lambda layers in model
+def _combine_r_and_d(x):
+    r, d = x
+    return r * (1. - d)
+
+
+class XPrizePredictor(object):
+    """
+    A class that computes a fitness for Prescriptor candidates.
+    """
+
+    def __init__(self, path_to_model_weights, data_url):
+        if path_to_model_weights:
+
+            # Load model weights
+            nb_context = 1  # Only time series of new cases rate is used as context
+            nb_action = len(NPI_COLUMNS)
+            self.predictor, _ = self._construct_model(nb_context=nb_context,
+                                                      nb_action=nb_action,
+                                                      lstm_size=LSTM_SIZE,
+                                                      nb_lookback_days=NB_LOOKBACK_DAYS)
+            self.predictor.load_weights(path_to_model_weights)
+
```

Comment:
Does not include option of downloading data on the fly, since the data is fixed. Instead, assert its existence.